parsers/nmparser/get_model_output.go

```diff
@@ -67,48 +68,137 @@ func GetModelOutput(filePath string, verbose bool, noExt bool, noGrd bool, noCov
 		}
 	}
 
+	epsCount := 0
+	for i := range results.ParameterStructures.Sigma {
+		if results.ParameterStructures.Sigma[i] > 0 {
+			epsCount++
+		}
+	}
+	etaCount := 0
+	for i := range results.ParameterStructures.Omega {
+		if results.ParameterStructures.Omega[i] > 0 {
+			etaCount++
+		}
+	}
+
+	if !noShk {
+		shkFilePath := strings.Join([]string{filepath.Join(dir, runNum), ".shk"}, "")
+		shkLines, err := utils.ReadLines(shkFilePath)
+		if err != nil {
+			panic(err)
+		}
+		results.ShrinkageDetails = ParseShkData(ParseShkLines(shkLines), etaCount, epsCount)
+	}
+
+	setDefaultValues(&results, etaCount, epsCount)
+	return results
+}
+
+func setDefaultValues(results *ModelOutput, etaCount, epsCount int) {
```

Comment:
this definitely needs tests to make sure the logic is behaving correctly

Reply:
yes, a unit test is required